_posts/2021-05-14-bring-gitops-to-your-functions-with-argocd-part-2-the-missing-piece-tekton.md

```diff
@@ -0,0 +1,612 @@
+CD installation namespace:
+
+```bash
+# Port-forward the ArgoCD API server
+$ kubectl port-forward svc/argocd-server -n argocd 8443:443
+Forwarding from 127.0.0.1:8443 -> 443
+Forwarding from [::1]:8443 -> 443
+
+# Get the password
+$ PASS=$(kubectl get secret argocd-initial-admin-secret \
+  -n argocd \
+  -o jsonpath="{.data.password}" | base64 -d)
+$ echo $PASS
+
+# Or log in:
+$ argocd login --name local 127.0.0.1:8443 --insecure \
+ --username admin \
+ --password $PASS
+WARNING: server certificate had error: x509: certificate signed by unknown authority. Proceed insecurely (y/n)? y
+Username: admin
+Password:
+'admin:login' logged in successfully
+Context 'local' updated
+ ```
+
+To get more detail about login process, please refer to
+this [link](https://argoproj.github.io/argo-cd/getting_started/#4-login-using-the-cli).
+
+Verify if you log into ArgoCD properly before moving on the next step.
+
+```bash
+$ argocd account list
+NAME   ENABLED  CAPABILITIES
+admin  true     login
+
+$ argocd app list
+NAME               CLUSTER                         NAMESPACE  PROJECT  STATUS  HEALTH   SYNCPOLICY  CONDITIONS  REPO                                                 PATH       TARGET
+app-of-apps        https://kubernetes.default.svc  openfaas   default  Synced  Healthy  Auto-Prune  <none>      https://github.com/developer-guy/hello-function.git  apps       master
+hello-function     https://kubernetes.default.svc  default    default  Synced  Healthy  Auto-Prune  <none>      https://github.com/developer-guy/hello-function.git  manifests  master
+openfaas-operator  https://kubernetes.default.svc  openfaas   default  Synced  Healthy  Auto-Prune  <none>      https://openfaas.github.io/faas-netes                           7.2.8                       7.2.8
+```
+
+Verify if everything is working before moving onto the next step.
+
+```bash
+$ kubectl get pods --namespace default
+NAME                             READY   STATUS             RESTARTS   AGE
+hellofunction-7849c4cf8d-zdgm4   0/1     InvalidImageName   0          9m30s
+# this is totally fine becuase we are using a placeholder as image name.
+
+$ kubectl get pods --namespace openfaas
+NAME                                 READY   STATUS    RESTARTS   AGE
+alertmanager-795797ffd4-8wq54        1/1     Running   0          12m
+basic-auth-plugin-85d885557c-422lz   1/1     Running   0          12m
+gateway-548fcb58-wcs4h               2/2     Running   1          12m
+nats-5fdf6476f-l6fz8                 1/1     Running   0          12m
+prometheus-7f48755f4f-5cslr          1/1     Running   0          12m
+queue-worker-5cfd6689f5-qtghb        1/1     Running   1          12m
+```
+
+### 3. Deploy Tekton and Tekton Trigger
+
+There are various ways to install _Tekton_ and _Tekton Trigger_ Trigger, one of them is _Tekton Operator_, and the other one is just with plain YAML manifest. We are going to deploy Tekton and Tekton Trigger with plain YAML manifest in this section.
```

Comment:
This line isn't adding anything because you don't compare or explain the various parts or ways.

In which case you could remove it completely and just say which away you are going to install Tekton, and link to their docs.